### PR TITLE
[CNXC-305] Filter results by MRN on backend

### PR DESCRIPTION
**Pull request by janakitti**
## Problem Statement
In the current implementation, the MRN filter feature does a client-side filter on results that are on the current page of the table. Ideally, it should be filtering through all results that belong to the user using a backend query parameter.

## Implementation
When the `pl-dircopy` instance is created, it is provided the Patient MRN as a `title`. This allows the UI to filter by MRN by passing the MRN into the `name` query parameter when calling `getFeeds` to fetch results to display on the table. 
In order to reduce the number of calls made to the backend, lodash `debounce` was added to debounce the PastAnalysisTable filter state updates.

### CODE REVIEW

File: src/assets/scss/components/seriesTable.scss [hunk 1]
```diff
@@ -11,3 +11,8 @@
     color: var(--pf-global--danger-color--100);
     margin-left: 0.5em;
 }
+
+.results-spinner-container {
+    display: flex;
+    justify-content: center;
+}
```
Comment (boris-nguyen) [inline src/assets/scss/components/seriesTable.scss:18]: @janakitti can you do me a favor and refactor this file to have 2 space indentation so it's consistent with the rest of the codebase.

File: src/shared/utils.tsx [hunk 2]
```diff
@@ -16,9 +16,9 @@ export const formatTime = (oldDay: string): string => {
   return oldDay.split('.')[0]
 }
 
-export const modifyDatetime = (oldDay: string): string => {
+export const modifyDatetime = (oldDay: number): string => {
```
Comment (boris-nguyen) [inline src/shared/utils.tsx:19]: Please, revert these changes since the logic of this function has been replaced with Luxon in [PR#38](https://github.com/darwinai/covidnet_ui/pull/51/files#). Also, please double check that the Luxon logic doesn't break the logic needed for this feature.

Comment (janakitti) [reply src/shared/utils.tsx:19]: This change was actually merged in from https://github.com/darwinai/covidnet_ui/pull/53. Will address this change when  working on implementing suggestions on that branch.

File: src/components/pastAnalysis/PastAnalysisTable.tsx [hunk 3]
```diff
@@ -43,23 +49,26 @@ const INITIAL_TABLE_STATE: TableState = {
...
 }
 
 enum TableReducerActions {
   UPDATE_MAX_FEED_ID = "UPDATE_MAX_FEED_ID",
   ADD_NEW_PAGE = "ADD_NEW_PAGE",
   INCREMENT_PAGE = "INCREMENT_PAGE",
   DECREMENT_PAGE = "DECREMENT_PAGE",
-  UPDATE_PLUGINS = "UPDATE_PLUGINS"
+  UPDATE_PLUGINS = "UPDATE_PLUGINS",
+  setFilter = "SET_FITLER"
```
Comment (boris-nguyen) [inline src/components/pastAnalysis/PastAnalysisTable.tsx:62]:
```suggestion
  SET_FILTER = "SET_FITLER"
```

Comment (janakitti) [reply src/components/pastAnalysis/PastAnalysisTable.tsx]: Done: https://github.com/darwinai/covidnet_ui/pull/51/commits/6d1f95724577add75d4554cfe4354a4eec62fb0d

**Review by boris-nguyen — APPROVED** [review]

**Review by colinywu — APPROVED** [review]
Great work Janakitti!
